URLSearchParams should be iterable

<!-- BUGS: Please use this template. -->
<!-- QUESTIONS: This is not a general support forum! Ask Qs at http://stackoverflow.com/questions/tagged/typescript -->
<!-- SUGGESTIONS: See https://github.com/Microsoft/TypeScript-wiki/blob/master/Writing-Good-Design-Proposals.md -->

**TypeScript Version:**  2.2.2

according to https://url.spec.whatwg.org/#urlsearchparams
URLSearchParams is iterable, but it is not iterable in typescript now.

this code failed to compile but run successful as plain JavaScript in google chrome.
**Code**

```ts
// A *self-contained* demonstration of the problem follows...
const params = new URLSearchParams()
params.set('k', 'v')
for(let [key, value] of params) {
    console.log(key, value)
}
```

**Expected behavior:**
compile successfully
**Actual behavior:**
error TS2488: Type must have a '[Symbol.iterator]()' method that returns an iterator.

--- a/src/lib/dom.iterable.d.ts
+++ b/src/lib/dom.iterable.d.ts
@@ -70,3 +70,10 @@ interface NodeListOf<TNode extends Node> {
 
     [Symbol.iterator](): IterableIterator<TNode>;
 }
+
+interface URLSearchParams {
+    /**
+     * iterate over key/value pairs
+     */
+    [Symbol.iterator](): IterableIterator<[string, string]>;
+}

--- a/src/lib/dom.iterable.d.ts
+++ b/src/lib/dom.iterable.d.ts
@@ -73,6 +73,18 @@ interface NodeListOf<TNode extends Node> {
 
 interface URLSearchParams {
     /**
+     * Returns an array of key, value pairs for every entry in the search params
+     */
+    entries(): IterableIterator<[string, string]>;
+    /**
+     * Returns a list of keys in the search params
+     */
+    keys(): IterableIterator<string>;
+    /**
+     * Returns a list of values in the search params
+     */
+    values(): IterableIterator<string>;
+    /**
      * iterate over key/value pairs
      */
     [Symbol.iterator](): IterableIterator<[string, string]>;

